### PR TITLE
Encrypted realm  / Multi wallet rudiments

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: app/src/main/kotlin/vergecurrency/vergewallet/service/model/DataManager.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    fun getAllEncryptedPreferences(context: Context): HashMap<UUID, String> {
+        val existingWalletFiles: HashMap<UUID, String> = HashMap<UUID, String>();
+        val datDir = context.dataDir.listFiles();
+        if (datDir != null) {
+            for (dataDir in datDir) {
+                if (dataDir.absolutePath.endsWith("vergecurrency.vergewallet/shared_prefs")) {
+                    val encryptedSharedPreferences = dataDir.listFiles();
+                    if (encryptedSharedPreferences != null && encryptedSharedPreferences.isNotEmpty()) {
+                        for (pref in encryptedSharedPreferences) {
+                            if (WalletDataIdentifierUtils.isEncryptedSharedPreferences(pref.name)) {
```
Comment (marpme) [inline app/src/main/kotlin/vergecurrency/vergewallet/service/model/DataManager.kt:30]: Convert into early returns
 ```java
if(datdir == null) { return ... }
```

File: app/src/main/kotlin/vergecurrency/vergewallet/service/model/DataManager.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        getOrCreateVergeDataRealm(walletId)
+    }
+
+
+    //Return identifier and wallet name of all existing encryptedSharedPreferences
+    fun getAllEncryptedPreferences(context: Context): HashMap<UUID, String> {
+        val existingWalletFiles: HashMap<UUID, String> = HashMap<UUID, String>();
+        val datDir = context.dataDir.listFiles();
+        if (datDir != null) {
+            for (dataDir in datDir) {
```
Comment (marpme) [inline app/src/main/kotlin/vergecurrency/vergewallet/service/model/DataManager.kt:25]: Stream over datdir and use functionalities like map and reduce to get this working. It's barely readable with having 8 if indentations

File: app/src/main/kotlin/vergecurrency/vergewallet/viewmodel/PinVerificationViewModel.kt [hunk 3]
```diff
@@ -6,7 +6,7 @@ import vergecurrency.vergewallet.service.model.EncryptedPreferencesManager
 class PinVerificationViewModel : ViewModel() {
 
 
-    fun setPin(pin: String) {
-        EncryptedPreferencesManager.pin = pin
+    fun setPin(pin: CharArray) {
+        //EncryptedPreferencesManager.pin = pin
```
Comment (marpme) [inline app/src/main/kotlin/vergecurrency/vergewallet/viewmodel/PinVerificationViewModel.kt:10]: Remove dead code.

File: app/src/main/kotlin/vergecurrency/vergewallet/viewmodel/ShowPaperkeyViewModel.kt [hunk 4]
```diff
@@ -5,15 +5,15 @@ import vergecurrency.vergewallet.service.model.EncryptedPreferencesManager.Compa
 import vergecurrency.vergewallet.service.model.MnemonicManager
 
 class ShowPaperkeyViewModel : ViewModel() {
-    var seed: Array<String>
+    var seed: Array<ByteArray>
```
Comment (marpme) [inline app/src/main/kotlin/vergecurrency/vergewallet/viewmodel/ShowPaperkeyViewModel.kt:8]: That's an array inside an array, why?

File: app/src/main/kotlin/vergecurrency/vergewallet/wallet/TransactionManager.kt [hunk 5]
```diff
@@ -1,11 +1,11 @@
 package vergecurrency.vergewallet.wallet
 
-import vergecurrency.vergewallet.service.repository.db.TransactionDAO
 
-class TransactionManager(private val transactionDAO: TransactionDAO) {
+class TransactionManager() {
 
     fun hasTransactions(): Boolean {
-        return transactionDAO.hasRecords();
+        //transactionDAO.hasRecords()
```
Comment (marpme) [inline app/src/main/kotlin/vergecurrency/vergewallet/wallet/TransactionManager.kt:7]: Fix or remove dead code.

File: app/src/main/kotlin/vergecurrency/vergewallet/wallet/WalletClient.kt [hunk 6]
```diff
@@ -125,9 +125,9 @@ class WalletClient : WalletClientInterface {
                 try {
                     val walletId = WalletId.decode(data)
 
-                    EncryptedPreferencesManager.walletId = walletId.identifier
-                    EncryptedPreferencesManager.walletName = walletName
-                    EncryptedPreferencesManager.walletSecret = buildSecret(walletId.identifier!!)
+                    EncryptedPreferencesManager.walletId = walletId.identifier!!.toByteArray()
```
Comment (marpme) [inline app/src/main/kotlin/vergecurrency/vergewallet/wallet/WalletClient.kt:128]: dirty `!!` add a check if you are unsure and handle it properly.

**Review by marpme — CHANGES_REQUESTED** [review]
Wow. these are way too many change inside of one single PR. 
I have simply see too much changes that were nearly unrelated to the realm switch.

File: app/src/main/kotlin/vergecurrency/vergewallet/viewmodel/ShowPaperkeyViewModel.kt (hunk 4 above)
Comment (ghost) [reply app/src/main/kotlin/vergecurrency/vergewallet/viewmodel/ShowPaperkeyViewModel.kt:8]: mnemonic words

Comment (marpme) [reply app/src/main/kotlin/vergecurrency/vergewallet/viewmodel/ShowPaperkeyViewModel.kt:8]: ah you have an array for all the words and then we have an array inside of it for each char byte encoded. Alright.

**Pull request closed without merging**